src/webui/abstractwebapplication.cpp

```diff
@@ -310,10 +310,30 @@ void AbstractWebApplication::increaseFailedAttempts()
 
 bool AbstractWebApplication::isAuthNeeded()
 {
-    return (env_.clientAddress != QHostAddress::LocalHost
-            && env_.clientAddress != QHostAddress::LocalHostIPv6
-            && env_.clientAddress != QHostAddress("::ffff:127.0.0.1"))
-            || Preferences::instance()->isWebUiLocalAuthEnabled();
+    qDebug("Checking auth rules against client address %s", qPrintable(env_.clientAddress.toString()));
+    if (isLANAddress(env_.clientAddress))
+        return Preferences::instance()->isWebUiLANAuthEnabled();
+    if (isLoopbackAddress(env_.clientAddress))
+        return Preferences::instance()->isWebUiLocalAuthEnabled();
+    return true;
+}
+
+bool AbstractWebApplication::isLoopbackAddress(QHostAddress clientAddress)
+{
+    return clientAddress == QHostAddress::LocalHost
+            || clientAddress == QHostAddress::LocalHostIPv6
+            || clientAddress == QHostAddress("::ffff:127.0.0.1");
+}
+
+bool AbstractWebApplication::isLANAddress(QHostAddress clientAddress)
+{
+    return clientAddress.isInSubnet(QHostAddress::parseSubnet("10.0.0.0/8"))
+            || clientAddress.isInSubnet(QHostAddress::parseSubnet("172.16.0.0/12"))
+            || clientAddress.isInSubnet(QHostAddress::parseSubnet("192.168.0.0/16"))
+            || clientAddress.isInSubnet(QHostAddress::parseSubnet("::ffff:a00:0/104")) // 10.0.0.0/8 mapped to ipv6
```

Comment:
[0a00](https://www.ultratools.com/tools/ipv6CIDRToRangeResult?ipAddress=%3A%3Affff%3A0a00%3A0%2F104) and [a00](https://www.ultratools.com/tools/ipv6CIDRToRangeResult?ipAddress=%3A%3Affff%3Aa00%3A0%2F104) are equivalent.